src/Umbraco.Infrastructure/Sync/DatabaseServerMessenger.cs

```diff
@@ -123,10 +125,12 @@ protected void Boot()
             // the service will *not* be able to properly handle our notifications anymore
             const int weight = 10;
 
-            if (!(_runtime is RuntimeState runtime))
-                throw new NotSupportedException($"Unsupported IRuntimeState implementation {_runtime.GetType().FullName}, expecting {typeof(RuntimeState).FullName}.");
 
-            var registered = runtime.MainDom.Register(
+            //TODO Why do we have interface if we expect to be exact type!!!?
```

Comment:
No idea about this one, prob best to just remove the comment and do what makes sense.